### DISCUSSION TITLE
internationalization: allow more flexibility for keys recognition

### DISCUSSION THREAD

**Discussion by Infini-Creation** [godotengine/godot-proposals#7261] [GUI]
It would be great if Godot would be able to recognize keys from translation files as patterns into a text and not just doing a very basic, quite binary, pattern replacement as it is now, by that I mean it is ok to have [KEY] but not [  KEY  !] (brackets are the boundaries of textfield in the editor).

Pattern matching are very advanced and can do a lot better than that (I've worked a lot with Perl for the last to decades and witnesses how its awesome regex features have spread to all other major languages while keeping the relative easiness to use them.)

If you have some text you want to split and add signs, they have to be embedded into translated text OR added as separate labels, which both not very great ways to do it, need more work and create a mess with containers.

**Comment by Calinou** [comment]
If you need to insert [placeholders](https://docs.godotengine.org/en/latest/tutorials/i18n/internationalizing_games.html#placeholders) in localization strings, you should use `String.format()`. Do not rely on concatenating several translated strings together, as this prevents translators from reordering placeholders (which is often required to have natural-sounding translations).

**Comment by Infini-Creation** [comment]
ok but that's not even the point, as it is now, the whole auto translation system is very difficult to use without help from code, as its purpose looks like to be fully managed by non-developers and without code changes.

And _RichTextLabel_ are not "compatible" at all (a single _bbcode_ tag kills translation ! I don't even consider a long text full of features)
